redwood-protocol-compose/src/commonMain/kotlin/app/cash/redwood/protocol/compose/ProtocolMismatchHandler.kt

```diff
@@ -48,5 +48,23 @@ public interface ProtocolMismatchHandler {
         )
       }
     }
+
+    /** A [ProtocolMismatchHandler] which prints out info for all callbacks. */
+    @JvmField
+    public val Logging: ProtocolMismatchHandler = object : ProtocolMismatchHandler {
```

Comment:
I'm tempted to say that instead of being a companion property this should be a nested class with a signature like
```kotlin
class Logging(logger: (String) -> Unit) : ProtocolMismatchHandler
```
which would allow the `Throwing` instance to be declared as
```
val Throwing = Logging { throw it }
```
and thus de-duplicate the string rendering code.